src/Helsenorge.Registries/CollaborationProtocolRegistry.cs

```diff
@@ -91,6 +91,15 @@ public async Task<CollaborationProtocolProfile> FindProtocolForCounterpartyAsync
             }
             catch (FaultException<CPAService.GenericFault> ex)
             {
+                if (_settings.ThrowMessageIfNoCpp)
```

Comment:
This is a harder hitting fix, but could in theory break messaging for some users.